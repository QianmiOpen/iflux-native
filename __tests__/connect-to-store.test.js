import React, { Component } from 'react'
import renderer from 'react-test-renderer';
import Store from '../lib/store'
import connectToStore from '../lib/connect-to-store'
// jest.mock('react-dom')


const appStore = Store({
  name: 'iflux'
})

appStore.init = function () {
  appStore.cursor().set('name', 'iflux-native');
};

appStore.setName = function (name) {
  appStore.cursor().set('name', name)
}

class HelloApp extends Component {
  componentDidMount() {
    appStore.init()
  }

  render() {
    const {store} = this.props;

    return (
      <div>{store.get('name')}</div>
    )
  }
}

const TestApp = connectToStore(appStore)(HelloApp);

describe('connectToStore test suite', () =>
  it('it initial render && store set name', () => {
    let Component = renderer.create(<TestApp />)
    expect(Component.toJSON()).toMatchSnapshot();

    appStore.setName('iflux2')
    expect(Component.toJSON()).toMatchSnapshot()
  })
)