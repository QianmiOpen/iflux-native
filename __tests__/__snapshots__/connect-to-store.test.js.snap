// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connectToStore test suite it initial render && store set name 1`] = `
<div>
  iflux-native
</div>
`;

exports[`connectToStore test suite it initial render && store set name 2`] = `
<div>
  iflux2
</div>
`;
